### ISSUE TITLE
MPI_UNWEIGHTED error

### ISSUE THREAD

**Issue by Aditya62** [firemodels/fds#6052]
I am trying to run the example file "activate_vents.fds" using the compiled version of fds (which I downloaded from the website recently).
When I try to submit the job on a queue system, I run into "MPI_UNWEIGHTED" error (attached:MPI_unweighted_error)
I have attached the script file used to submit the job on the queue (attached: pbs1.sh).
I have also included the input file(activate_vents.fds) that I am using.
The command I use to submit the job is "run_script_mpi pbs1.sh".

[Attachments.zip](https://github.com/firemodels/fds/files/1745837/Attachments.zip)

**Comment by sbenkorichi** [comment]
Ok,
Thanks, I will take a look

**Comment by mcgratta** [comment]
The script `pbs1.sh` refers to mpich2. FDS was not compiled with mpich2. I would not expect this to work.

**Comment by sbenkorichi** [comment]
Yes, as I expected it. The error of MPI_UNWEIGHTED is usually coming from openmpi (which what you are using). 
If you want to use fds bundle that is downloaded from the web page. Then, you don't have to use openmpi, we already provide mpiexec (mpi intel ) included. So, when you install fds, you simply run it as: `$ mpiexec -np X fds job.fds `

If you want to use job scheduler, then install torque since you are working with Ubuntu. 
Here is an example on how we submit a job.
```
#!/bin/bash
#PBS -N job_name
#PBS -e /home/userid/.../job_name.err
#PBS -o /home/userid/.../job_name.log
#PBS -l nodes=2:ppn=2
#PBS -l walltime=2:0:0
export OMP_NUM_THREADS=1
cd /home/userid/.../
mpiexec -np 4 /home/userid/FDS/FDS6/bin/fds job_name.fds
```
Full details can be found here: 
https://github.com/firemodels/fds/wiki/Installing-and-Running-FDS-on-a-Linux-Cluster

-- On the other hand, if you are planning to compile fds yourself, and then create a module for it (although you still can use module with option above). Then, you need to compile openmpi,
you would still have the option to run it either with mpiexec or mpirun (but making sure to provide the full path to them).

**Comment by Aditya62** [comment]
I downloaded the code from the GitHub site and compiled it using the "make
mpi_intel_linux_64" command. It successfully compiled but it gave me
warning while it was compiling.
[image: Inline image 2]
I ran the example "activate_vents.fds" and ran into a Segmentation Fault[image:
Inline image 1]

**Comment by sbenkorichi** [comment]
Please use the github to upload the images, you are replying from email, and this doesn't show the figures from my end.

**Comment by Aditya62** [comment]
I downloaded the code from the GitHub site and compiled it using the "make mpi_intel_linux_64" command. It successfully compiled but it gave me warning while it was compiling. 
![screenshot from 2018-02-21 18-32-37](https://user-images.githubusercontent.com/36547341/36513678-645b04d2-1736-11e8-92cc-872e8b48954c.png)

I ran the example "activate_vents.fds" and ran into a Segmentation Fault
![screenshot from 2018-02-21 18-33-46](https://user-images.githubusercontent.com/36547341/36513695-7928b81e-1736-11e8-923f-504df0897adf.png)

**Comment by sbenkorichi** [comment]
Before, you go into compiling fds, how you are compiling openmpi?
What intel version do you use?

**Comment by sbenkorichi** [comment]
I found this post, I wrote a while ago, although it might need some updating. 
https://www.sbenkorichi.com/fds-compilation-linux/

1. Use latest openmpi (3) 
2. compile impi_intel_linux_64  as explained here: https://github.com/firemodels/fds/wiki/FDS-Compilation

**Comment by Aditya62** [comment]
I am using intel compiler openmpi/1.10.2-intel-pmi2.
I'll try and use openmpi (3) and compile again.

Thank you

**Comment by sbenkorichi** [comment]
Well,
you are not obliged to compile openmpi and fds to start using parallel jobs. Unless, you are working in research. If you are just want to use parallel jobs, then you can use the mpi provided with fds. 
Check how you can use module with it here: 
https://github.com/firemodels/fds/wiki/Modules

**Comment by Aditya62** [comment]
Okay, I recently started working on my research and hence I am first trying to run the given examples using the most recent version of fds.  
So here's the update: I was able to upgrade to openmpi ver 2, (i was limited because i do not have privileges  to install version 3 on the cluster). I changed the makefile from "mpiifort" to "mpif90" and  compiled using "impi_intel_linux_64" . The compilation was successful and the example files run fine now.
Thank you for your suggestions.
